### DISCUSSION TITLE
Upgrade guidance for OSS to Pro

### DISCUSSION THREAD

**Discussion by patkujawa-wf** [jobrunr/jobrunr#439] [Q&A] [answered]
We tried upgrading from v4 (OSS version) to v5 (pro) and ended up in a state where I had to drop the jobrunr tables in order to downgrade/revert and successfully start our service. I had made the necessary code changes (in our case, that was just `RetryFilter.DEFAULT_NBR_OF_RETRIES` -> `DefaultRetryFilter.DEFAULT_NBR_OF_RETRIES`) but it seems like there are some migrations (we're running AWS Aurora MySQL) that were changed as well.

~~I looked at the breaking changes in https://github.com/jobrunr/jobrunr/releases/tag/v5.0.0 but it wasn't clear to me that the db schema migrations themselves changed (if I'm understanding correctly).~~ 

EDIT: I think this is just a difference between OSS and Pro versions, since there's no difference between these:
* https://github.com/jobrunr/jobrunr/tree/v4.0.10/core/src/main/resources/org/jobrunr/storage/sql/common/migrations
* https://github.com/jobrunr/jobrunr/tree/v5.0.1/core/src/main/resources/org/jobrunr/storage/sql/common/migrations

But I can see the migration names and numbers differ for Pro v5.0.3:
![image](https://user-images.githubusercontent.com/6391742/166068959-a9e64811-3704-4487-9897-b85560d1ef3f.png)

Other than stopping my service (finishing or aborting all in-flight jobs) and dropping all the existing jobrunr tables, is there a path to migrate to v5/pro gracefully?

**Comment by rdehuyss** [comment, marked as answer]
Have a try with JobRunr Pro 5.1.0-RC1

**Reply by patkujawa-wf** [reply]
Thanks, that seems to have upgraded cleanly for us! That said, since we were using job metadata, I may want to upgrade to v5 OSS and fix that part of the code, **then** upgrade to Pro, if that'll provide a safer upgrade route. I'll have to experiment a bit.
